### PR TITLE
allow users to use node port for istio ingress without producing another service for lb

### PR DESCRIPTION
**Pull request by linsun**
and specify lb ip for istio gateway.

use consistent config with istio ingress gateway.

### CODE REVIEW

File: install/kubernetes/helm/istio/values.yaml [hunk 1]
```diff
@@ -93,10 +93,15 @@ ingress:
...
     loadBalancerIP: ""
-    nodePort:
-      enabled: false
-      port: 32000
-
+    type: LoadBalancer #change to NodePort, ClusterIP or LoadBalancer if need be
+    ports:
+    - port: 80
+      name: http
+      nodePort: 32000
```
Comment (costinm) [inline install/kubernetes/helm/istio/values.yaml]: Is it ok to add nodePort with LoadBalancer ? If yes, great. Otherwise - we need extra filtering.

**Comment by costinm** [comment]
But please fix the duplicated IP.

File: install/kubernetes/helm/istio/values.yaml (hunk 1 above)
Comment (linsun) [reply install/kubernetes/helm/istio/values.yaml]: I'll do more testing thanks.

**Comment by linsun** [comment]
@costinm for duplicate ip, i was hoping each chart (istio-ingress and ingress gateway) will pull its own LB ip.  I'll test it.

**Comment by linsun** [comment]
re duplicate ip, @costin, i don't think that is an issue.  For example, i set  the following in values.yaml

```
ingress:
  enabled: true
  serviceAccountName: default
  autoscaleMin: 1
  autoscaleMax: 1
  resources: {}
# limits:
#  cpu: 100m
#  memory: 128Mi
# requests:
#  cpu: 100m
#  memory: 128Mi
  service:
    loadBalancerIP: "1.1.1.1"
    type: LoadBalancer #change to NodePort, ClusterIP or LoadBalancer if need be
    ports:
    - port: 80
      name: http
      nodePort: 32000
    - port: 443
      name: https
    selector:
      istio: ingress
#
# ingressgateway configuration
#
ingressgateway:
  enabled: true
  serviceAccountName: istio-ingressgateway-service-account
  autoscaleMin: 1
  autoscaleMax: 1
  resources: {}
# limits:
#  cpu: 100m
#  memory: 128Mi
# requests:
#  cpu: 100m
#  memory: 128Mi
  service:
    name: istio-ingressgateway #DNS addressible
    labels:
      istio: ingressgateway
    #namespace: istio-system
    loadBalancerIP: "2.2.2.2"
```

and below is the output after helm template:

```
# Source: istio/charts/ingress/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: istio-ingress
  namespace: istio-system
  labels:
    chart: ingress-0.8.0
    release: RELEASE-NAME
    heritage: Tiller
    istio: ingress
spec:
  loadBalancerIP: "1.1.1.1"
  type: NodePort
  selector:
    istio: ingress
  ports:
    -
      name: http
      nodePort: 32000
      port: 80
    -
      name: https
      port: 443
---

---
# Source: istio/charts/ingressgateway/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: istio-ingressgateway
  namespace: istio-system
  labels:
    chart: ingressgateway-0.8.0
    release: RELEASE-NAME
    heritage: Tiller
    istio: ingressgateway
spec:
  loadBalancerIP: "2.2.2.2"
  type: LoadBalancer
  selector:
    istio: ingressgateway

```

File: install/kubernetes/helm/istio/values.yaml (hunk 1 above)
Comment (linsun) [reply install/kubernetes/helm/istio/values.yaml]: yes, it does work.  see below:

```

~/Downloads/istio-release-0.8-20180511-16-55/install/kubernetes ⌚ 21:54:35
$ cat istio-ingress-test.yaml 
apiVersion: v1
kind: Service
metadata:
  name: istio-ingress
  namespace: istio-system
  labels:
    chart: ingress-0.8.0
    release: RELEASE-NAME
    heritage: Tiller
    istio: ingress
spec:
  # loadBalancerIP: "1.1.1.1"
  type: LoadBalancer 
  selector:
    istio: ingress
  ports:
    -
      name: http
      nodePort: 32000
      port: 80
    -
      name: https
      port: 443


$ kubectl get services -n istio-system
NAME                       TYPE           CLUSTER-IP       EXTERNAL-IP    PORT(S)                                                               AGE
istio-citadel              ClusterIP      172.21.124.13    <none>         8060/TCP,9093/TCP                                                     6h
istio-egressgateway        ClusterIP      172.21.126.139   <none>         80/TCP,443/TCP                                                        6h
istio-ingress              LoadBalancer   172.21.117.218   169.60.83.11   80:32000/TCP,443:32745/TCP                                            6s


```

**Pull request merged**